engine/robot.py

```diff
@@ -102,6 +104,9 @@ def __init__(self, x_pos, y_pos, heading, epsilon, max_v, radius, is_sim=True, p
         self.gyro_rotation = [0, 0, 0]  # TEMPORARY
         self.linear_v = 0
         self.angular_v = 0
+        if not self.is_sim:
```

Comment:
do we still want to initialize one when we don't have a simulation?

Reply:
why are we only initializing one if its not a simulation, but then later in robot.py we only use the motorcontroller if it IS a simulation?